### PR TITLE
Implement ICard and IDeck interfaces

### PR DESCRIPTION
**Pull request by matyimat**
in addition i added BaseCards to the GameState for Marci

### CODE REVIEW

File: Dixit_Logic/Classes/Card.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        /// <summary>
+        ///  The card indentifier
+        /// </summary>
+        private int _id;
+
+        /// <summary>
+        /// The image of the card
+        /// </summary>
+        Bitmap _image;
```
Comment (SilverGeri) [inline Dixit_Logic/Classes/Card.cs]: There is no use having this field anymore, Bitmap cannot be used as source, we need an IValueConverter for that, but I already implemented one that can make ImageSource from an ID in https://github.com/VSZM/Dixit_By_MonkeyzWithKeyboardz_ELTE/pull/42

It is unnecessary to contain the images in Dixit_Logic when id is enough

Comment (matyimat) [reply Dixit_Logic/Classes/Card.cs]: ok, I'm going to correct this tomorrow

File: Dixit_Logic/Classes/Deck.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+        /// <summary>
+        /// Construct a deck from the given cards.
+        /// </summary>
+        /// <param name="cards">An IList implementation what contains card(ICard)s</param>
+        public Deck(IList<ICard> cards)
+        {
+            _cards = new List<ICard>();
+
+            foreach (ICard card in cards)
```
Comment (VSZM) [inline Dixit_Logic/Classes/Deck.cs]: Use List.AddRange  method here. 
https://msdn.microsoft.com/en-us/library/z883w3dc(v=vs.110).aspx

File: Dixit_Logic/Classes/Deck.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        }
+
+        /// <summary>
+        /// That is the only way to put cards in the deck after the Deck 
+        /// object is constructed.
+        /// </summary>
+        /// <param name="cards">An IList implementation what contains card(ICard)s</param>
+        public void AddCards(IList<ICard> cards)
+        {
+            foreach (ICard card in cards)
```
Comment (VSZM) [inline Dixit_Logic/Classes/Deck.cs]: https://msdn.microsoft.com/en-us/library/z883w3dc(v=vs.110).aspx

File: Dixit_Logic/Classes/Hand.cs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+        /// <summary>
+        /// Construct a hand from the given cards.
+        /// </summary>
+        /// <param name="cards">An IList implementation what contains card(ICard)s</param>
+        public Hand(IList<ICard> cards)
+        {
+            _cards = new List<ICard>();
+
+            foreach (ICard card in cards)
```
Comment (VSZM) [inline Dixit_Logic/Classes/Hand.cs]: http://stackoverflow.com/questions/3173320/text-progress-bar-in-the-console

File: Dixit_Logic/Classes/Hand.cs [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+                _cards.Add(card);
+            }
+        }
+
+        /// <summary>
+        /// That is the only way to put cards in the hand after the Hand 
+        /// object is constructed.
+        /// </summary>
+        /// <param name="cards">An IList implementation what contains card(ICard)s</param>
+        public void AddCards(IList<ICard> cards)
```
Comment (VSZM) [inline Dixit_Logic/Classes/Hand.cs:53]: I think there should be a method which adds a Single ICard. In each turn the player plays a Card, and then Draws a single one. With this method only you will have to wrap these single values into Lists.

**Review by VSZM — CHANGES_REQUESTED** [review]

File: Dixit_Logic/Classes/Hand.cs (hunk 4 above)
Comment (matyimat) [reply Dixit_Logic/Classes/Hand.cs]: I suppose you wanted to signify the same problem as previous two : https://msdn.microsoft.com/en-us/library/z883w3dc(v=vs.110).aspx

File: Dixit_Logic/Classes/Hand.cs (hunk 5 above)
Comment (matyimat) [reply Dixit_Logic/Classes/Hand.cs:53]: It is also true. I ll add an AddCard method

File: Dixit_Logic/Classes/Hand.cs (hunk 4 above)
Comment (VSZM) [reply Dixit_Logic/Classes/Hand.cs]: Yes, I mixed up a link with another one on my clipboard : )

**Review by VSZM — APPROVED** [review]

**Pull request merged**